source/binding/Bitstream.cpp

```diff
@@ -622,13 +700,76 @@ static bool unpackConcatenation(const StreamingConcatenationExpression& lhs, Pac
             ASSERT(iterConcat == packed.cend() && !bit);
         }
         else {
-            auto rvalue = unpackBitstream(*stream->type, iter, iterEnd, bitOffset, dynamicSize);
+            auto& arrayType = *operand.type;
+            const Type* elemType = nullptr;
+            ConstantRange with;
+            if (stream->with) {
+                auto range = Bitstream::evaluateWith(arrayType, *stream->with, context);
+                if (!range)
+                    return false;
+                with = *range;
+                elemType = arrayType.getArrayElementType();
+                ASSERT(elemType != nullptr);
+                if (dynamicSize > 0 && !stream->with->width) {
+                    auto withSize = elemType->bitstreamWidth() * with.width();
+                    if (withSize >= dynamicSize)
+                        dynamicSize = 0;
+                    else
+                        dynamicSize -= withSize;
+                }
+            }
+            auto rvalue = unpackBitstream(
+                !stream->with
```

Comment:
This expression is too complicated to pass as a parameter, can you pull it out into its own temporary variable?